app/main.py

```diff
@@ -1,25 +1,66 @@
 from abc import ABC
+from typing import Any, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.protected_name = "_" + name
+        self.public_name = name
+
+    def __get__(self, instance: Any, owner: Any) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Any, value: int) -> None:
+        if value < self.min_amount or value > self.max_amount:
```

Comment:
it should be `and`

Reply:
there cannot be "and" here! how can the value be less than the minimum and more than the maximum at the same time?